common/libs/network_protocol/network_protocol.proto

```diff
@@ -0,0 +1,130 @@
+syntax = "proto2";
```

Comment:
At the risk of punishing a good deed, now might be a good time to write some comments explaining what each of these messages is, since I presume you have it in your head?